core/src/main/java/org/wildfly/channel/VersionResult.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2023 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.channel;
+
+import java.util.Objects;
+
+/**
+ * The result of version search.
+ */
+public class VersionResult {
+    private String version;
+    private String channelName;
+
+    public VersionResult(String version, String channelName) {
+        Objects.requireNonNull(version);
+        Objects.requireNonNull(version);
+
+        this.version = version;
+        this.channelName = channelName;
+    }
+
+    /**
+     * The discovered artifact version.
+     *
+     * @return
+     */
+    public String getVersion() {
+        return version;
+    }
+
+    /**
+     * The name of the channel the artifact was found in.
+     * @return
+     */
+    public String getChannelName() {
```

Comment:
could you make it return `Optional<String>` instead?

Reply:
Does the Optional make sense for VersionResult? The `MavenArtifact` might be resolved using 'resolveDirectMavenArtifact' bypassing channels, but I think the version query would always have to be resolved against one of the channel

Reply:
@spyrkob that makes sense. So I suppose that one of the `Objects.requireNonNull` above was for the `channelName`, right?

Reply:
yes it was, I must have copy/pasted the line and forgot to change the parameter name, sorry

Reply:
I realised that's not correct after all. The channel name is not mandatory in the specification, so in both `VersionResult` and `MavenArtifact` the channelName can be null even if the artifact/version was resolved from one.

I'll update the PR with the change